riff-raff/app/notification/DeployFailureNotifications.scala

```diff
@@ -50,36 +47,44 @@ class DeployFailureNotifications(config: Config,
-
-    val deriveAnghammaradTargets = for {
-      yaml <- targetResolver.fetchYaml(parameters.build)
-      deployGraph <- Resolver.resolveDeploymentGraph(yaml, deploymentTypes, magenta.input.All).toEither
-    } yield {
-      TargetResolver.extractTargets(deployGraph).toList.flatMap { target =>
-        val maybeAccountId = uuid.flatMap(uuid => getAwsAccountIdTarget(target, parameters, uuid)).toList
-        List(App(target.app), Stack(target.stack), Stage(parameters.stage.name)) ++ maybeAccountId
+  def getTargets(uuid: UUID, parameters: DeployParameters): List[Target] = {
+    val attemptToDeriveTargets = for {
+        yaml <- targetResolver.fetchYaml(parameters.build)
+        deployGraph <- Resolver.resolveDeploymentGraph(yaml, deploymentTypes, magenta.input.All).toEither
+      } yield {
+        TargetResolver.extractTargets(deployGraph).toList.flatMap { target =>
+          val maybeAccountId = getAwsAccountIdTarget(target, parameters, uuid).toList
+          List(App(target.app), Stack(target.stack), Stage(parameters.stage.name)) ++ maybeAccountId
+        }
       }
+    attemptToDeriveTargets match {
+      case Right(targets) => targets
+      case Left(_) => riffRaffTargets
     }
+  }
 
-    deriveAnghammaradTargets match {
-      case Right(targets) =>
-        log.info(s"Sending anghammarad notification with targets: ${targets.toSet}")
-        val failureMessage = s"${parameters.deployer.name} for ${parameters.build.projectName} (build ${parameters.build.id}) to stage ${parameters.stage.name} failed."
-        Anghammarad.notify(
-          subject = s"${parameters.deployer.name} failed",
-          message = failureMessage,
-          sourceSystem = "riff-raff",
-          channel = All,
-          target = targets,
-          // TODO: Figure out the correct action for scheduled deploys that don't start
-          actions = uuid.map(id => List(Action("View failed deploy", url(id)))).getOrElse(Nil),
-          topicArn = anghammaradTopicARN,
-          client = snsClient
-        ).recover { case ex => log.error(s"Failed to send notification (via Anghammarad) for $uuid", ex) }
-      case Left(_) =>
-        log.error(s"Failed to derive targets required to notify about failed scheduled deploy: $uuid")
-    }
+  def notifyViaAnghammarad(notificationContents: NotificationContents, targets: List[Target]) = {
+    log.info(s"Sending anghammarad notification with targets: ${targets.toSet}")
+    Anghammarad.notify(
+      subject = notificationContents.subject,
+      message = notificationContents.message,
+      sourceSystem = "riff-raff",
+      channel = All,
+      target = targets,
+      actions = notificationContents.actions,
+      topicArn = anghammaradTopicARN,
+      client = snsClient
+    ).recover { case ex => log.error(s"Failed to send notification (via Anghammarad)", ex) }
+  }
+
+  def deployUnstartedNotification(error: ScheduledDeployNotificationError): Unit = {
```

Comment:
Would `scheduledDeployFailureNotification` be a more explicit name here?

Reply:
I think the names you've suggested are an improvement. I've updated them here: https://github.com/guardian/riff-raff/pull/614/commits/5dd6bea2d47343ba8f500e70482f3c823f621cf0.